ts/store/transforms/dateISO8601Tranform.ts

```diff
@@ -19,7 +20,7 @@ const dateReviver = (_: any, value: any): any => {
     typeof value === "string" &&
     value.match(/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}/)
   ) {
-    return new Date(value);
```

Comment:
All the function block can be replaced with:

```javascript
const dateReviver = (_: any, value: any): any => {
  const decodedValue = DateFromISOString.decode(value);
  return decodedValue.isRight() ? decodedValue.value : value;
};
```
This because the **DateFromISOString** takes care about checking if the given value can be encoded as a valid Date object, so we can avoid string and format checks